jerry-core/jcontext/jcontext.h

```diff
@@ -56,9 +72,17 @@ typedef struct jerry_context_data_header
  * The purpose of this header is storing
  * all global variables for Jerry
  */
-typedef struct
+struct jerry_context_t
 {
-  /* Update JERRY_CONTEXT_FIRST_MEMBER if the first member changes */
+  /* The value of external context members must be preserved across initializations and cleanups. */
+#ifdef JERRY_ENABLE_EXTERNAL_CONTEXT
+#ifndef JERRY_SYSTEM_ALLOCATOR
+  jmem_heap_t *heap_p; /**< point to the heap aligned to JMEM_ALIGNMENT. */
```

Comment:
I would put this before or after the context, probably after would be better, since the starting offset would be fixed. Then we don't need this pointer.

Reply:
The instance-based approach also contained this pointer, at the end of the instance struct, even if the heap always started right after the instance. I'd keep this pointer as it allows for future developments, e.g., one that allows placing context structures and heaps in different memory regions (which is already available for the builtin global context approach but is not available for external contexts right now).